plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -902,11 +902,35 @@ func PodFitsHostPorts(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *s
 	}
 
 	existingPorts := nodeInfo.UsedPorts()
-	for wport := range wantPorts {
-		if wport != 0 && existingPorts[wport] {
-			return false, []algorithm.PredicateFailureReason{ErrPodNotFitsHostPorts}, nil
+
+	// try to see whether two hostPorts will conflict or not
+	for existingPort := range existingPorts {
+		existingHostPortInfo := decode(existingPort)
+
+		if existingHostPortInfo.hostIP == schedutil.DefaultBindAllHostIP {
+			// try to detect if there exists a conflict
+			if portConflict(existingHostPortInfo, wantPorts) {
+				return false, []algorithm.PredicateFailureReason{ErrPodNotFitsHostPorts}, nil
+			}
 		}
 	}
+
+	for wantPort := range wantPorts {
+		wantHostPortInfo := decode(wantPort)
+
+		if wantHostPortInfo.hostIP == schedutil.DefaultBindAllHostIP {
+			// try to detect if there exists a conflict
+			if portConflict(wantHostPortInfo, existingPorts) {
+				return false, []algorithm.PredicateFailureReason{ErrPodNotFitsHostPorts}, nil
+			}
+		} else {
+			// general check hostPort conflict procedure for hostIP is not 0.0.0.0
+			if wantPort != "" && existingPorts[wantPort] {
+				return false, []algorithm.PredicateFailureReason{ErrPodNotFitsHostPorts}, nil
+			}
+		}
+	}
+
```

Comment:
Two loops is OK for me, while I think you can wrap them into a method and call it here.

Reply:
hmm, good point.